source/Halibut.TestProxy/TcpTunnel.cs

```diff
@@ -40,6 +40,10 @@ public async Task Tunnel(CancellationToken cancellationToken)
             {
                 await fromWriter.CompleteAsync();
                 await toWriter.CompleteAsync();
+                fromStream.Close();
+                toStream.Close();
```

Comment:
Consider using that `Try` class here. Since if `fromStrean.Close()` throws the other close methods will not be called.